packages/dcos-integration-test/extra/test_tweeter_demo.py

```diff
@@ -0,0 +1,103 @@
+import logging
+import os
+import time
+
+import pytest
+import requests
+import retrying
+
+import test_helpers
+
+from test_packaging import (
+    skipif_insufficient_resources
+)
+
+log = logging.getLogger(__name__)
+
+DEMO_RESOURCE_REQUIREMENTS = {
+    'number_of_nodes': 5,
+    'node': {
+        'disk': 0,
+        'mem': 0,
+        'cpus': 1
+    }
+}
+
+
+def _get_tweeter_app_definition():
+    response = requests.get('https://raw.githubusercontent.com/mesosphere/tweeter/master/tweeter.json')
+    return response.json()
+
+
+def _get_post_tweets_app_definition():
+    response = requests.get('https://raw.githubusercontent.com/mesosphere/tweeter/master/post-tweets.json')
+    return response.json()
+
+
+def _get_tweet_count(address):
+    """Count the number of tweets on the tweeter app's webpage
+
+    Args:
+        address: str URL
+    """
+    response = requests.get(address)
+    return response.text.count('tweet-content')
+
+
+@retrying.retry(
+    wait_fixed=10000,
+    stop_max_delay=60000)
+def _wait_for_tweet_count(url):
+    # This is 25 tweets because of how paging in tweeter is defined
+    assert _get_tweet_count(url) == 25, "Did not find expected 25 tweets"
+
+
+@pytest.mark.skipif(
+    test_helpers.expanded_config.get('security') == 'strict',
+    reason='This test requires specific service accounts if security is strict. ' +
+    'See corresponding test_tweeter_demo in dcos-enterprise.')
```

Comment:
Did you address the comment against `reason=` here?
Is`security=strict` *required* for this test?

Reply:
This specific test should be skipped if it is in strict mode, hence the `skipif(...('security') == 'strict'`. There is a corresponding version in dcos-enterprise which does the tweeter demo including the security configuration. I will update it to `"There is a corresponding test test_tweeter_demo_enterprise which is instead run when security==strict"`

Reply:
Ah! I see. The reason is actually giving a pointer to the EE test. It was not obvious and it was confusing initially.

The `reason=` explanation  now sounds good to me. Thank you for the clarification.